plugin-maven/src/main/java/com/tisonkun/os/maven/RepositorySessionInjector.java

```diff
@@ -28,7 +30,24 @@ static void injectRepositorySession(Logger logger, MavenSession session, Map<Str
                 final Field f = cls.getDeclaredField("systemProperties");
                 f.setAccessible(true);
                 repoSessionProps = (Map<String, String>) f.get(repoSession);
-                repoSessionProps.putAll(dict);
+                try {
+                    repoSessionProps.putAll(dict);
+                } catch (Exception ex2) {
+                    // In Maven 4, DefaultCloseableSession uses an immutable map
```

Comment:
I test locally that only with this change (i.e., no DetectPropertyContributor and the disbale thing), it seems this plugin can work properly also.

Reply:
Yes, it can be simplified.  If needed, we may want to release a Maven 3 plugin and a Maven 4 plugin separately. I've raised https://github.com/tisonkun/os-detector/pull/9, but feel free to cherry-pick to this PR (I could not push to it).